src/AutoMapper/MapperConfiguration.cs

```diff
@@ -34,49 +26,42 @@ public class MapperConfiguration : IConfigurationProvider, IConfiguration
         private readonly ConcurrentDictionary<TypePair, CreateTypeMapExpression> _typeMapExpressionCache =
             new ConcurrentDictionary<TypePair, CreateTypeMapExpression>();
 
-        internal readonly ConcurrentDictionary<string, IProfileExpression> _formatterProfiles =
-            new ConcurrentDictionary<string, IProfileExpression>();
+        private readonly ConcurrentDictionary<string, Profile> _formatterProfiles =
+            new ConcurrentDictionary<string, Profile>();
 
         private Func<Type, object> _serviceCtor = ObjectCreator.CreateObject;
 
         private readonly List<string> _globalIgnore;
 
-        public MapperConfiguration(IEnumerable<IObjectMapper> mappers, IEnumerable<ITypeMapObjectMapper> typeMapObjectMappers)
+        public MapperConfiguration(Action<IConfiguration> configure) : this(configure, MapperRegistry.Mappers, TypeMapObjectMapperRegistry.Mappers)
+        {
+        }
+
+        public MapperConfiguration(Action<IConfiguration> configure, IEnumerable<IObjectMapper> mappers, IEnumerable<ITypeMapObjectMapper> typeMapObjectMappers) 
         {
             _typeMapFactory = new TypeMapFactory();
             _mappers = mappers;
             _typeMapObjectMappers = typeMapObjectMappers;
             _globalIgnore = new List<string>();
-        }
-
-        public MapperConfiguration() : this(MapperRegistry.Mappers, TypeMapObjectMapperRegistry.Mappers)
-        {
-        }
 
-        public MapperConfiguration(Action<IConfiguration> configure) : this(MapperRegistry.Mappers, TypeMapObjectMapperRegistry.Mappers)
-        {
             configure(this);
-        }
 
-        public MapperConfiguration(Action<IConfiguration> configure, IEnumerable<IObjectMapper> mappers, IEnumerable<ITypeMapObjectMapper> typeMapObjectMappers) 
-            : this(mappers, typeMapObjectMappers)
-        {
-            configure(this);
+            Seal();
+
+            ExpressionBuilder = new ExpressionBuilder(this);

```

Comment:
It should pay for use. Now it is, because the ExpressionBuilder constructor doesn't do much. But we should keep it that way or create it lazily.